Fix for #1119, thumb shows as a transparent block when deactivated in Lollipop+

josmas opened the pull request:
This only affects lollipop+ (API 21), and they added a method to stop splitting the track with the thumb drawable (active or inactive).

Tested also on 4.4.3, does not change behaviour (works fine there in first place; default _split track_ value is false).

The branch is based on ucr, not master. Please review.

josmas commented:
yeah, I see the same thing, which if you compare it with the screenshots in the issue, it is basically the same (removes one of the colours), but getting rid of the transparency bit.

I'll make an app with just a Seekbar and see if I get the same result; there's a bit of hack in there for `API level < 16`, which seems innocuous enough, but I'll check if it is doing anything... it might take a bit though (probably sometime during the weekend). I'll assign to myself.

josmas commented:
Hey @ewpatton I think I figured this out and has nothing to do with the fix, which is still valid.

For some reason, when the _Thumb_ is made disabled (transparent really), we also [disable the whole Seekbar](https://github.com/mit-cml/appinventor-sources/blob/master/appinventor/components/src/com/google/appinventor/components/runtime/Slider.java#L184). It does not make much sense to me, but that explains why all colour is drained.

So if that is correct, I guess this fix should be applied, but we should also remove the disabling of the slider; shall I go ahead and add that to this PR or you'd rather handle it as separate issues? I am just thinking if this would need any kind of _upgrader_ for existing projects, but it's just a change in the look and feel, not in functionality; what you think? shall I get the YA_VERSION changed or is the change too minimal?

ewpatton commented:
@josmas I would place them in the same change so that if we need to revert it for any reason both changes are undone together. Regarding YA_VERSION, I don't think this sufficiently changes the user-facing semantics to warrant an update. Rather, this change is just fixing an issue where the current behavior doesn't align with the expected semantics.

josmas commented:
@ewpatton, I have rebased on `ucr` and pushed.

Once the disable for the whole bar is removed, the touch listener keeps working even without the thumb, so I added the suggestions by @NmdOfficial and @kkashi01 in #1119, but still needs the _setSplitTrack_ method to fix the original issue.

josmas commented:
@ewpatton I have now! :)
I pushed my companion to the old emulator and the thumb does not go away (as expected, as the transparency is only 16+) but the Seekbar is not fully disabled and the touch events don't respond. And no crashes, which is the important part!
